services/worker/tests/job_runners/config/test_parquet_and_info.py

```diff
@@ -0,0 +1,552 @@
+    ) -> ConfigParquetAndInfoJobRunner:
+        return ConfigParquetAndInfoJobRunner(
+            job_info={
+                "type": ConfigParquetAndInfoJobRunner.get_job_type(),
+                "dataset": dataset,
+                "config": config,
+                "split": None,
+                "job_id": "job_id",
+                "force": force,
+                "priority": Priority.NORMAL,
+            },
+            app_config=app_config,
+            processing_step=ProcessingStep(
+                name=ConfigParquetAndInfoJobRunner.get_job_type(),
+                input_type="config",
+                requires=[],
+                required_by_dataset_viewer=False,
+                ancestors=[],
+                children=[],
+                job_runner_version=ConfigParquetAndInfoJobRunner.get_job_runner_version(),
+            ),
+            hf_datasets_cache=libraries_resource.hf_datasets_cache,
+            parquet_and_info_config=parquet_and_info_config,
+        )
+
+    return _get_job_runner
+
+
+def assert_content_is_equal(content: Any, expected: Any) -> None:
+    print(content)
+    assert set(content) == {"parquet_files", "dataset_info"}, content
+    assert content["parquet_files"] == expected["parquet_files"], content
+    assert len(content["dataset_info"]) == len(expected["dataset_info"]), content
+    content_value = content["dataset_info"]
+    expected_value = expected["dataset_info"]
+    assert set(content_value.keys()) == set(expected_value.keys()), content
+    for key in content_value.keys():
+        if key != "download_checksums":
+            assert content_value[key] == expected_value[key], content
+    assert len(content_value["download_checksums"]) == 1, content
+    content_checksum = list(content_value["download_checksums"].values())[0]
+    expected_checksum = list(expected_value["download_checksums"].values())[0]
+    assert content_checksum == expected_checksum, content
+
+
+def test_compute(
+    app_config: AppConfig,
+    get_job_runner: GetJobRunner,
+    parquet_and_info_config: ParquetAndInfoConfig,
+    hub_datasets: HubDatasets,
+) -> None:
+    dataset = hub_datasets["public"]["name"]
+    config = hub_datasets["public"]["config_names_response"]["config_names"][0]["config"]
+    job_runner = get_job_runner(dataset, config, app_config, parquet_and_info_config, False)
+    assert job_runner.process()
+    cached_response = get_response(kind=job_runner.processing_step.cache_kind, dataset=dataset, config=config)
+    assert cached_response["http_status"] == HTTPStatus.OK
+    assert cached_response["error_code"] is None
+    assert cached_response["job_runner_version"] == job_runner.get_job_runner_version()
+    assert cached_response["dataset_git_revision"] is not None
+    content = cached_response["content"]
+    assert len(content["parquet_files"]) == 1
+    assert_content_is_equal(content, hub_datasets["public"]["parquet_and_info_response"])
+
+
+def test_compute_legacy_configs(
```

Comment:
this checks that configs are not deleted when other configs of a dataset are pushed and that outdated configs are deleted.